dask/dataframe/io/tests/test_io.py

```diff
@@ -119,13 +119,18 @@ def check(i):
         t = bcolz.ctable(
             [[1, 2, 3], [1.0, 2.0, 3.0], ["a", "b", "a"]], names=["x", "y", "a"]
         )
-        d = dd.from_bcolz(t, chunksize=2)
+
+        with pytest.warns(FutureWarning):
+            d = dd.from_bcolz(t, chunksize=2)
```

Comment:
In addition to checking that a `FutureWarning` is raised, we should also check that the warning message is both what we're expecting and meaningful for users. `pytest.warns` has a nice `match=` keyword argument for checking against the warning message. 

Also, since there are several places where we call `from_bcolz` in tests and want to check the deprecation warning emitted, it's worth consolidating that logic into a single context manager. See `check_bcolz_deprecation_warning` in the diff below

```diff
diff --git a/dask/dataframe/io/tests/test_io.py b/dask/dataframe/io/tests/test_io.py
index 06f74008..f73e727f 100644
--- a/dask/dataframe/io/tests/test_io.py
+++ b/dask/dataframe/io/tests/test_io.py
@@ -1,3 +1,4 @@
+import contextlib
 from concurrent.futures import ThreadPoolExecutor
 from threading import Lock
 
@@ -112,6 +113,12 @@ def test_from_array_with_record_dtype():
     assert (d.compute().to_records(index=False) == x).all()
 
 
+@contextlib.contextmanager
+def check_bcolz_deprecation_warning():
+    with pytest.warns(FutureWarning, match="bcolz was deprecated"):
+        yield
+
+
 def test_from_bcolz_multiple_threads():
     bcolz = pytest.importorskip("bcolz")
 
@@ -120,7 +127,7 @@ def test_from_bcolz_multiple_threads():
             [[1, 2, 3], [1.0, 2.0, 3.0], ["a", "b", "a"]], names=["x", "y", "a"]
         )
 
-        with pytest.warns(FutureWarning):
+        with check_bcolz_deprecation_warning():
             d = dd.from_bcolz(t, chunksize=2)
 
         assert d.npartitions == 2
@@ -128,7 +135,7 @@ def test_from_bcolz_multiple_threads():
         assert list(d.x.compute(scheduler="sync")) == [1, 2, 3]
         assert list(d.a.compute(scheduler="sync")) == ["a", "b", "a"]
 
-        with pytest.warns(FutureWarning):
+        with check_bcolz_deprecation_warning():
             d = dd.from_bcolz(t, chunksize=2, index="x")
 
         L = list(d.index.compute(scheduler="sync"))
@@ -153,20 +160,22 @@ def test_from_bcolz():
         [[1, 2, 3], [1.0, 2.0, 3.0], ["a", "b", "a"]], names=["x", "y", "a"]
     )
 
-    with pytest.warns(FutureWarning):
+    with check_bcolz_deprecation_warning():
         d = dd.from_bcolz(t, chunksize=2)
-        assert d.npartitions == 2
-        assert is_categorical_dtype(d.dtypes["a"])
-        assert list(d.x.compute(scheduler="sync")) == [1, 2, 3]
-        assert list(d.a.compute(scheduler="sync")) == ["a", "b", "a"]
-        L = list(d.index.compute(scheduler="sync"))
-        assert L == [0, 1, 2]
-
+    assert d.npartitions == 2
+    assert is_categorical_dtype(d.dtypes["a"])
+    assert list(d.x.compute(scheduler="sync")) == [1, 2, 3]
+    assert list(d.a.compute(scheduler="sync")) == ["a", "b", "a"]
+    L = list(d.index.compute(scheduler="sync"))
+    assert L == [0, 1, 2]
+
+    with check_bcolz_deprecation_warning():
         d = dd.from_bcolz(t, chunksize=2, index="x")
-        L = list(d.index.compute(scheduler="sync"))
-        assert L == [1, 2, 3] or L == [1, 3, 2]
+    L = list(d.index.compute(scheduler="sync"))
+    assert L == [1, 2, 3] or L == [1, 3, 2]
 
-        # Names
+    # Names
+    with check_bcolz_deprecation_warning():
         assert sorted(dd.from_bcolz(t, chunksize=2).dask) == sorted(
             dd.from_bcolz(t, chunksize=2).dask
         )
@@ -176,9 +185,10 @@ def test_from_bcolz():
 
         dsk = dd.from_bcolz(t, chunksize=3).dask
 
-        t.append((4, 4.0, "b"))
-        t.flush()
+    t.append((4, 4.0, "b"))
+    t.flush()
 
+    with check_bcolz_deprecation_warning():
         assert sorted(dd.from_bcolz(t, chunksize=2).dask) != sorted(dsk)
 
 
@@ -190,7 +200,7 @@ def test_from_bcolz_no_lock():
         [[1, 2, 3], [1.0, 2.0, 3.0], ["a", "b", "a"]], names=["x", "y", "a"], chunklen=2
     )
 
-    with pytest.warns(FutureWarning):
+    with check_bcolz_deprecation_warning():
         a = dd.from_bcolz(t, chunksize=2)
         b = dd.from_bcolz(t, chunksize=2, lock=True)
         c = dd.from_bcolz(t, chunksize=2, lock=False)
@@ -212,7 +222,7 @@ def test_from_bcolz_filename():
         )
         t.flush()
 
-        with pytest.warns(FutureWarning):
+        with check_bcolz_deprecation_warning():
             d = dd.from_bcolz(fn, chunksize=2)
 
         assert list(d.x.compute()) == [1, 2, 3]
@@ -225,7 +235,7 @@ def test_from_bcolz_column_order():
         [[1, 2, 3], [1.0, 2.0, 3.0], ["a", "b", "a"]], names=["x", "y", "a"]
     )
 
-    with pytest.warns(FutureWarning):
+    with check_bcolz_deprecation_warning():
         df = dd.from_bcolz(t, chunksize=2)
 
     assert list(df.loc[0].compute().columns) == ["x", "y", "a"]
```

Reply:
Thanks for the review, James!